indexd_test_utils/__init__.py

```diff
@@ -149,8 +151,13 @@ def indexd_server():
     hostname = 'localhost'
     port = 8001
     debug = False
+    proc_handler = mp.Process
+    if six.PY3:
+        # Note: explicitly specifying fork, as spawn is the default in
+        # py3.8+ on macos.
+        proc_handler = mp.get_context("fork").Process
```

Comment:
It looks like this issue runs deeper.

The official python docs state that "fork" is considered unsafe on macOS:
https://docs.python.org/3.8/library/multiprocessing.html#contexts-and-start-methods

The change to make "spawn" the default for macOS is documented and discussed in this ticket:
https://bugs.python.org/issue33725

Some issues related to "spawn" on macOS are documented here:
https://bugs.python.org/issue40106

Could you investigate this further to see what our options are?

Btw, if we were to stick with using fork, we should also be explicit and check that the os is macOS.   https://docs.python.org/3/library/sys.html#sys.platform

Reply:
Other than the current solution, we could mock indexd. I am not sure how much of a concern this is given that it is only tests. This issue in mac isn't new https://bugs.python.org/msg344590, but macOS has become more aggressive at detecting this and crashing early. I've encountered this solution in other popular python libraries, including pytorch https://github.com/pytorch/pytorch/pull/36687 . 

Moreover, Werkzeug only provides support for fork https://github.com/pallets/werkzeug/blob/d781b06c3fbac8eaeb3bfb61bea005a3a5288b91/src/werkzeug/serving.py#L763 . After some thought, it is not clear there is a good way to get around this that doesn't require a large effort.

Reply:
Since this is only applicable to tests, we can table this for now and make a separate ticket to rewrite this section without using multiprocessing or something like that. I also think this might affect other projects by the time we update them to py38